src/JustSaying/Messaging/Channels/ConsumerBus.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using JustSaying.AwsTools.MessageHandling;
+using JustSaying.AwsTools.MessageHandling.Dispatch;
+using JustSaying.Messaging.Monitoring;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.Channels
+{
+    internal class ConsumerBus : IConsumerBus
+    {
+        private readonly IMultiplexer _multiplexer;
+        private readonly IList<IMessageReceiveBuffer> _buffers;
+        private readonly IList<IChannelConsumer> _consumers;
+        private readonly ILogger _logger;
+        private readonly IConsumerConfig _consumerConfig;
+
+        internal ConsumerBus(
+            IList<ISqsQueue> queues,
+            IConsumerConfig consumerConfig,
+            IMessageDispatcher messageDispatcher,
+            IMessageMonitor monitor,
+            ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<ConsumerBus>();
+            _consumerConfig = consumerConfig;
+
+            _multiplexer = new RoundRobinQueueMultiplexer(
```

Comment:
Should we DI-inject the multiplexer (or add a factory abstraction to create them) rather than hard-code it?